File: src/main/java/cal/fstg/java/controller/DocumentController.java
package cal.fstg.java.controller;

//import cal.fstg.java.model.Admin;
//import cal.fstg.java.model.Document;
//import cal.fstg.java.service.AdminService;
//import cal.fstg.java.service.DocumentService;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.web.bind.annotation.*;
//
//import java.util.List;
//import java.util.Optional;

//@RestController
//@RequestMapping("/documents")
//public class DocumentController {
//
//    private final DocumentService documentService;
//
//    @Autowired
//    public DocumentController(DocumentService documentService) {
//        this.documentService = documentService;
//    }
//
//    @GetMapping("/get/all")
//    public List<Document> getAll() {
//        return documentService.getAll();
//    }
//
//    @GetMapping("/get/{id}")
//    public Optional<Document> getOne(@PathVariable long id) {
//        return documentService.getOne(id);
//    }
//
//    @PostMapping("/save")
//    public void save(@RequestBody Document document) {
//        documentService.save(document);
//    }
//
//    @DeleteMapping("/delete/{id}")
//    public void delete(@PathVariable long id) {
//        documentService.delete(id);
//    }
//}

File: src/main/kotlin/cal/fstg/stagerkotlin/controller/DocumentController.kt
package cal.fstg.stagerkotlin.controller

import cal.fstg.stagerkotlin.model.Document
import cal.fstg.stagerkotlin.model.Employer
import cal.fstg.stagerkotlin.service.DocumentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/documents")
class DocumentController(@Autowired private val documentService: DocumentService) {

    @GetMapping("/get/all")
    fun getAll() = documentService.getAll()

    @GetMapping("/get/{id}")
    fun getOne(@PathVariable id: String) = documentService.getOne(id)

    @PostMapping("/save")
    fun save(@RequestBody document: Document) = documentService.save(document)

    @DeleteMapping("/delete/{id}")
    fun delete(@PathVariable id: String) = documentService.delete(id)
}